test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1044,6 +1051,16 @@ private static String getPendingClusterStateTasks() {
         return null;
     }
 
+    private static String getHotThreads() {
+        try {
+            Response response = adminClient().performRequest(new Request("GET", "/_nodes/hot_threads"));
+            return EntityUtils.toString(response.getEntity());
+        } catch (IOException e) {
+            fail(e, "Failed to retrieve hot threads in the cluster during cleanup");
```

Comment:
I suggest we don't throw AssertionError here which will cover up the original error. Log an error message should be fine.